stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/RetrofitCall.kt

```diff
@@ -40,83 +46,78 @@ internal class RetrofitCall<T : Any>(
         call.cancel()
     }
 
-    override fun execute(): Result<T> {
-        return execute(call)
-    }
+    override fun execute(): Result<T> = runBlocking { await() }
 
     override fun enqueue(callback: Call.Callback<T>) {
-        enqueue(call) {
-            if (!canceled.get()) {
-                callback.onResult(it)
+        scope.launch {
+            enqueue(call) { result ->
+                scope.launch { notifyResult(result) { callback.onResult(it) } }
             }
         }
```

Comment:
Hey @JcMinarro,

Since we already use `awaitResponse` which converts callback style into a suspendable coroutine I wonder if we can use `await` here instead of invoking `enqueue` of child call

Reply:
I agree. Retrofit provides some Kotlin extensions such as `awaitResponse`, so we can keep the Coroutines style with those extensions and `try-catch` expression.

Reply:
I am just simplifying it ;)